Fixes the model parser and tests in show version

LukeDRussell opened the pull request:
Signed-off-by: Luke Russell <lukedrussell+git@outlook.com>

pabelanger commented:
recheck

pabelanger commented:
Okay, now we are properly gating on parsing tests. you can now iterate properly on your patch

LukeDRussell commented:
recheck

LukeDRussell commented:
Done.

pabelanger commented on tests/parser_templates/cli/show_interfaces/main.yaml:
```diff
@@ -18,9 +18,9 @@
       - interfaces['Tunnel0']['admin-status'] == 'enabled'
       - interfaces['Tunnel0']['oper-status'] == 'down'
-      - interfaces['GigabitEthernet0']['counters']['in-broadcast-pkts'] == '4891144'
+      - interfaces['GigabitEthernet0']['counters']['in-broadcast-pkts'] | int == 4891144
```
There shouldn't be a need to cast here, if you revert this change, it will work.

pabelanger commented on tests/parser_templates/cli/show_interfaces/main.yaml:
```diff
@@ -18,9 +18,9 @@
+      - interfaces['GigabitEthernet0']['counters']['in-broadcast-pkts'] | int == 4891144
       - interfaces['GigabitEthernet0']['counters']['in-multicast-pkts'] == None
-      - interfaces['Virtual-Access1']['mtu'] == '1500'
+      - interfaces['Virtual-Access1']['mtu'] | int == 1500
```
same comment

LukeDRussell replied:
I think the output of these parsers should be integers for straight numbers. 

This may be a problem with `command_parser`, as I haven't been able to get it to return int types, even when using jinja2 filters.

I'll revert these for now to get the other stuff in, but these should be moved to integers at some point.

pabelanger reviewed (changes requested):
Thanks, if you want to address comments and rebase against current master, into single commit, should make for easier reviewing.

LukeDRussell committed:
Improves model regex, removes restart conditions
system.restart_time and system.restart_tz cause parser to hang if there is no restart line in show version
model regex was matching the incorrect line for certain IOS or IOS-XE models and versions

Signed-off-by: Luke Russell <lukedrussell+git@outlook.com>

pabelanger commented on tests/parser_templates/cli/show_version/16.6.4.txt:
```diff
@@ -64,19 +64,18 @@ Technology Package License Information:
 -----------------------------------------------------------------
 Technology    Technology-package           Technology-package
               Current       Type           Next reboot
-              ------------------------------------------------------------------
```
what's the reason for the formatting change here? I am guessing copypasta issue on original commit?

LukeDRussell replied:
Yes, it was a copy paste problem.

pabelanger commented on tests/parser_templates/cli/show_version/main.yaml:
```diff
@@ -1,4 +1,22 @@
 ---
+### IOS 15.1(4)M4
```
do you mind moving this back to original location, this helps keep the diff smaller and easier to review.

LukeDRussell replied:
The tests are in ascending order now, which I think is more important than diff size. Hopefully there will be more test samples from others in the future.
